frideos_library/lib/blocs/todos_bloc.dart

```diff
@@ -0,0 +1,167 @@
+import 'package:todos_repository_core/todos_repository_core.dart';
+
+import 'package:frideos/frideos_dart.dart';
+
+import 'package:frideos_library/models/models.dart';
+
+class TodosBloc {
+  final TodosRepository repository;
+
+  TodosBloc({this.repository}) {
+    print('-------TODOS BLOC INIT--------');
+    _init();
+  }
+
+  _init() async {
+    // Load the todos from the repository
+    await loadTodos();
+
+    // Listening for changes in the todos list, updated the streams and save
+    // to the repository. It is the same as:
+    //
+    // todosItems.outStream.listen((todos) => onTodosChange(...)
+    //
+    todosItems.onChange((todos) => onTodosChange(
+        allComplete, hasCompletedTodos, todos, onDataLoaded, todosSender));
+
+    // Listening for changes in the VisibilityFilter and filter the visible
+    // todos depending on the current filter.:
+    activeFilter
+        .onChange((filter) => onFilterChange(todosItems, visibleTodos, filter));
+  }
+
+  // When the data is loaded this function will save the todos, update the visible
+  // todos, and send to the statsBloc the todos list.
+  Function onDataLoaded;
+
+  // STREAMS
+  //
+  final todosItems = StreamedList<Todo>();
+  final visibleTodos = StreamedList<Todo>();
+  final activeFilter =
+      StreamedValue<VisibilityFilter>(initialData: VisibilityFilter.all);
+  final currentTodo = StreamedValue<Todo>(initialData: Todo('Initializing'));
+  final allComplete = StreamedValue<bool>(initialData: false);
+  final hasCompletedTodos = StreamedValue<bool>(initialData: false);
+
+  // SENDER (to send the todosItems list to the StatsBloc)
+  //
+  final todosSender = ListSender<Todo>();
+
+  // METHODS
+  //
+  void loadTodos() async {
+    var todos = await repository.loadTodos();
+    todosItems.value = todos?.map(Todo.fromEntity).toList() ?? [];
+    todosSender.send(todosItems.value);
+
+    onDataLoaded = () {
+      saveTodos();
+      updateVisibleItems();
+      todosSender.send(todosItems.value);
+    };
+  }
+
+  void updateVisibleItems() =>
+      visibleTodos.value = filterTodos(todosItems.value, activeFilter.value);
+
+  void saveTodos() => repository
+      .saveTodos(todosItems.value.map((item) => item.toEntity()).toList());
+
+  void addTodo(Todo todo) => todosItems.addElement(todo);
+
+  void updateTodo(Todo todo) {
+    todosItems.replace(currentTodo.value, todo);
+    currentTodo.value = todo;
+  }
+
+  void addedit(bool isEditing, String task, String note) {
```

Comment:
Perhaps `addEdit`? I initially read this as "added it".

Reply:
Yes, absolutely. I'll change it :-)